Source/WebCore/platform/network/ResourceHandle.cpp

```diff
@@ -137,9 +139,11 @@ void ResourceHandle::failureTimerFired()
 
 void ResourceHandle::loadResourceSynchronously(NetworkingContext* context, const ResourceRequest& request, StoredCredentialsPolicy storedCredentialsPolicy, SecurityOrigin* sourceOrigin, ResourceError& error, ResourceResponse& response, Vector<uint8_t>& data)
 {
-    if (auto constructor = builtinResourceHandleSynchronousLoaderMap().get(request.url().protocol().toStringWithoutCopying())) {
-        constructor(context, request, storedCredentialsPolicy, error, response, data);
-        return;
+    if (auto protocol = request.url().protocol(); !protocol.isEmpty()) {
+        if (auto constructor = builtinResourceHandleSynchronousLoaderMap().get(protocol.toAtomString())) {
```

Comment:
I suggest we use `toExistingAtomString` here instead. Since it’s only used one other place we might think about changing its return type, though.

Reply:
I am adopting toExistingAtomString() but not changing the return type in this patch because I worry this will blow up the patch size some more. Will look into it in a follow up.